### PR TITLE
Add YML files for issue templates

### LINKED ISSUE #6: Add issue templates
For instance, check https://github.com/stevemao/github-issue-templates/tree/master/.github/ISSUE_TEMPLATE but much shorter.

Probably something for

- Bug reports
- Feature request

### PR DESCRIPTION
**Pull request by datejada**
# Pull request details

## Describe the changes made in this pull request

Add template files for reporting issues for bugs and features. In addition, blank issues are not allowed anymore using the config.yml

## List of related issues or pull requests

Closes #6 

## Collaboration confirmation

As a contributor I confirm

- [x] I read and followed the instructions in README.dev.md
- [NA] The documentation is up to date with the changes introduced in this Pull Request (or NA)
- [x] Tests are passing
- [x] Lint is passing

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: "💡 Feature Request"
+description: Create a new ticket for a new feature request
+title: "💡 [REQUEST] - <title>"
+labels: [
+  "question"
+]
+body:
+  - type: input
```
Comment (abelsiqueira) [inline .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:8]: I think this does not apply

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      required: true
+  - type: textarea
+    id: drawbacks
+    attributes:
+      label: "Drawbacks"
+      description: What are the drawbacks/impacts of your feature request ?
+      placeholder: Identify the drawbacks and impacts while being neutral on your feature request
+    validations:
+      required: true
+  - type: textarea
```
Comment (abelsiqueira) [inline .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:40]: Maybe don't need this? The pr and issue are linked if the issue is added to the pr

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Hi Diego, thanks for the PR. Should we also use one for opportunities/technical debt? I am not sure where to put improve documentation/ci. Maybe an exploration issue template as well for things like testing different models, or benchmarking. We can discuss with the group if you agree.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  - type: textarea
+    id: reprod
+    attributes:
+      label: "Reproduction steps"
+      description: Please enter an explicit description of your issue
+      value: |
+        1. Go to '...'
+        2. Click on '....'
+        3. Scroll down to '....'
+        4. See error
```
Comment (clizbe) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:25]:
```suggestion
        4. [Specific error message]
```

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+name: "💡 Feature Request"
+description: Create a new ticket for a new feature request
+title: "💡 [REQUEST] - <title>"
```
Comment (clizbe) [inline .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:3]:
```suggestion
title: "<title>"
```

**Review by clizbe — CHANGES_REQUESTED** [review]
Good first draft!
Please also add a template for issues of "Type: improvement"
(Should be very similar to "Type: addition")
Do not include [IMPROVEMENT] in the title - we just want that for Bugs.

**Comment by clizbe** [comment]
@datejada So you know, this is reviewed and waiting for changes. :)

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      placeholder: "#Issues IDs"
+    validations:
+      required: false
+  - type: textarea
+    id: unresolved_question
+    attributes:
+      label: "Unresolved questions"
+      description: What questions still remain unresolved ?
+      placeholder: Identify any unresolved issues.
+    validations:
```
[the remark this answers is not in the document]
Comment (datejada) [reply .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:62]: Ok! I agree, let's start simple, and we will see what we need to add throughout the project.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      description: Please enter an explicit description of your issue
+      placeholder: Short and explicit description of your incident...
+    validations:
+      required: true
+  - type: textarea
+    id: reprod
+    attributes:
+      label: "Reproduction steps"
+      description: Please enter an explicit description of your issue
+      value: |
```
[the remark this answers is not in the document]
Comment (datejada) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml:21]: I changed steps 2 and 3 to be more related to out workflow

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        1. Go to '...'
+        2. Click on '....'
+        3. Scroll down to '....'
+        4. See error
+      render: bash
+    validations:
+      required: true
+  - type: textarea
```
[the remark this answers is not in the document]
Comment (datejada) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml:29]: Yes, I agree that the entire error message is more meaningful at this stage than a screenshot.

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml (hunk 2 above)
Comment (datejada) [reply .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:40]: I agree, it has been removed

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml (hunk 1 above)
Comment (datejada) [reply .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:8]: I agree; It has been removed

**Comment by abelsiqueira** [comment]
Looks like you updated this the wrong way. The branch now contains parts of main, instead of point on top of main

**Comment by datejada** [comment]
Yes, sorry @abelsiqueira, I did something wrong in the last update :/ 
Do you have any suggestions on how to fix it? 
Thanks!

**Comment by abelsiqueira** [comment]
I would try the following:

1. Backup
2. Create a squash commit (see hackmd)
3. Rebase that

I think you shouldn't even have conflicts, but check the final result to see if it makes sense. Let me know if it doesn't work.

**Comment by datejada** [comment]
Hi @abelsiqueira, I think I solved it. I reset my branch hard, then push --force to go back to the previous state, and then go over again on the procedure of rebasing to the main branch. I think it is okay now. Please double-check. Thanks!

**Comment by abelsiqueira** [comment]
It worked

**Comment by abelsiqueira** [comment]
> Good first draft! Please also add a template for issues of "Type: improvement" (Should be very similar to "Type: addition") Do not include [IMPROVEMENT] in the title - we just want that for Bugs.

@datejada, I think you didn't to this one? Maybe the name FEATURE_REQUEST should change accordingly as well?

**Comment by datejada** [comment]
Yes, I only went through the comments in the files; I forgot to look at the comment in the PR. Thanks! I'll add the changes.

File: .github/ISSUE_TEMPLATE/ADDITION-REQUEST.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
+name: "💡 Feature Request"
+description: Create a new ticket for a new feature request
```
Comment (clizbe) [inline .github/ISSUE_TEMPLATE/ADDITION-REQUEST.yml:2]:
```suggestion
name: "Addition Request"
description: Create a ticket for a new addition (or feature) request
```

Comment (clizbe) [reply .github/ISSUE_TEMPLATE/ADDITION-REQUEST.yml:2]: I would prefer not to have emojis on the Issues page, except maybe for bugs...

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 9]
```diff
@@ -0,0 +1,48 @@
+name: "🐛 Bug Report"
+description: Create a new ticket for a bug.
+title: "🐛 [BUG] - <title>"
```
Comment (clizbe) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:3]:
```suggestion
name: "🐛 Bug Report"
description: Create a new ticket for a bug.
title: "[BUG] - <title>"
```

Assuming that the title is what's shown on the Issues page.

File: .github/ISSUE_TEMPLATE/IMPROVEMENT-REQUEST.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
+name: "💡 Improvement Request"
+description: Create a new ticket for a new improvement request
```
Comment (clizbe) [inline .github/ISSUE_TEMPLATE/IMPROVEMENT-REQUEST.yml:2]:
```suggestion
name: "Improvement Request"
description: Create a ticket to request an improvement of something that already exists.
```
No emojis please and changed the description.

**Review by clizbe — CHANGES_REQUESTED** [review]
Looks good - just my little OCD details. :)

**Review by clizbe — APPROVED** [review]
Looks good! Thanks for handling my pickiness.

**Review by abelsiqueira — APPROVED** [review]
LGTM, thanks!

**Pull request merged**